### PR TITLE
Add a sample for Spring Boot REST API

### PR DESCRIPTION
**Pull request by yomalip**
## Purpose
A sample Spring Boot REST API for a Reading List is required for the following task : https://github.com/wso2-enterprise/choreo/issues/23217

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

## Samples
A sample Spring Boot REST API with the following endpoints:
- GET /books: Retrieves a list of all books in the database.
- GET /books/{id}: Retrieves a specific book identified by its unique ID.
- POST /books: Creates a new book and adds it to the database.
- PUT /books: Updates an existing book identified by its ID provided in the request body.
- DELETE /books/{id}: Deletes a specific book from the database.

### CODE REVIEW

File: echo-service/readme.md
[the remark this answers is not in the document]
Comment (yomalip) [reply echo-service/readme.md:1]: Same as above

File: java-product-catalog-rest-api/readme.md [hunk 1]
```diff
@@ -19,8 +19,8 @@ Please refer to the Choreo documentation under the [Develop a REST API](https://
 #### Use the following build config when creating this component in Choreo:
 
 - Build Preset: **Dockerfile**
```
Comment (chameerar) [inline java-product-catalog-rest-api/readme.md:21]: Could you please change this to build pack?

Comment (yomalip) [reply java-product-catalog-rest-api/readme.md:21]: Will do chameera

**Review by chameerar — APPROVED** [review]
LGTM

**Pull request merged**